biz.aQute.bndlib/src/aQute/bnd/service/result/Private.java

```diff
@@ -0,0 +1,11 @@
+package aQute.bnd.service.result;
+
+import aQute.lib.exceptions.Exceptions;
+
+public class Private {
```

Comment:
This class should look like:

```java
class Exceptions {
	static RuntimeException duck(Throwable t) {
		throwsUnchecked(t);
		throw new AssertionError("unreachable");
	}

	@SuppressWarnings("unchecked")
	static <E extends Throwable> void throwsUnchecked(Throwable throwable) throws E {
		throw (E) throwable;
	}
}
```

And then aQute.bnd.service.result.FunctionWithException will call the `duck` method in this class. This way the entire `aQute.bnd.service.result` package is disconnected from the `aQute.lib.exceptions` package. The `aQute.bnd.service.result` package has its own private copy of `duck`.